### PR TITLE
[Network] `az network vnet`: Add `--enable-encryption` parameter to support enable encryption on virtual network

### PR DESCRIPTION
**Pull request by wangzelin007**
**Description**

[issue link](https://github.com/Azure/azure-cli/issues/20033)

Add `--enable-encryption` parameter to az network vnet create.
Add `--enable-encryption` parameter to az network vnet update.
Add `--encryption-enforcement-policy` parameter to az network vnet create.
Add `--encryption-enforcement-policy` parameter to az network vnet update.
`az network nic create/update/show/list` can display the **vnetEncryptionSupported** flag
`az network vnet peering create/update/list/show` can display the **remoteVirtualNetworkEncryption** info
**Testing Guide**

`azdev test test_network_vnet_with_encryption --live`

**History Notes**

[Network] `az network vnet create`: Add `--enable-encryption` argument to support enable encryption on virtual network
[Network] `az network vnet update`: Add `--enable-encryption` argument to support enable encryption on virtual network
[Network] `az network vnet create`: Add `--encryption-enforcement-policy` argument to choose If Virtual Machine without encryption is allowed in encrypted Virtual Network.
[Network] `az network vnet update`: Add `--encryption-enforcement-policy` argument to choose If Virtual Machine without encryption is allowed in encrypted Virtual Network.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -6467,11 +6468,21 @@ def create_vnet(cmd, resource_group_name, vnet_name, vnet_prefixes='10.0.0.0/16'
     if bgp_community is not None and cmd.supported_api_version(min_api='2020-06-01'):
         VirtualNetworkBgpCommunities = cmd.get_models('VirtualNetworkBgpCommunities')
         vnet.bgp_communities = VirtualNetworkBgpCommunities(virtual_network_community=bgp_community)
+    if enable_encryption is not None:
+        if not vnet.encryption:
+            vnet.encryption = {}
```
Comment (dtuCloud) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:6473]: with this block, then line 6476 would never be hit?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -6467,11 +6468,21 @@ def create_vnet(cmd, resource_group_name, vnet_name, vnet_prefixes='10.0.0.0/16'
...
+        vnet.encryption["enabled"] = enable_encryption
+    if encryption_enforcement_policy is not None:
+        if not vnet.encryption:
```
Comment (dtuCloud) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:6476]: so probably here we should check if "enabled" is set in vnet.encryption?

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 1 above)
Comment (wangzelin007) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:6473]: The line 6474 will ensure that vnet.encryption = {"enabled": true|false}.
And vnet.encryption can be none sometimes, 'enabled' in vnet.encryption will raise TypeError: argument of type 'NoneType' is not iterable.

**Comment by wangzelin007** [comment]
azp /run

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**